presto-main/src/test/java/com/facebook/presto/operator/scalar/TestMathFunctions.java

```diff
@@ -245,12 +263,29 @@ public void testFloor()
         assertFunction("floor(-123.0)", DOUBLE, -123.0);
         assertFunction("floor(123.45)", DOUBLE, 123.0);
         assertFunction("floor(-123.45)", DOUBLE, -124.0);
+
         assertFunction("floor(REAL '123.0')", REAL, 123.0f);
         assertFunction("floor(REAL '-123.0')", REAL, -123.0f);
         assertFunction("floor(REAL '123.45')", REAL, 123.0f);
         assertFunction("floor(REAL '-123.45')", REAL, -124.0f);
-        assertFunction("floor(CAST(NULL as DOUBLE))", DOUBLE, null);
+
+        assertFunction("floor(DECIMAL '123')", createDecimalType(3), SqlDecimal.of("123"));
```

Comment:
Split these tests into 3 sections (separated by empty lines), each covering one of short-short, long-short, long-long.